android_webview/test/shell/tango/jni/AnchorManager.cpp

```diff
@@ -18,31 +18,44 @@ std::shared_ptr<Anchor> AnchorManager::addAnchor(double timestamp,
   std::shared_ptr<Anchor> anchor(new Anchor(timestamp,
                                             cameraModelMatrix,
                                             anchorModelMatrix));
+  anchorsMutex.lock();
   anchors[anchor->getIdentifier()] = anchor;
+  anchorsMutex.unlock();
   return anchor;
 }
 
 void AnchorManager::removeAnchor(uint32_t identifier) {
+  anchorsMutex.lock();
   std::unordered_map<uint32_t, std::shared_ptr<Anchor>>::iterator it = 
```

Comment:
this is a good time to use auto. Alternatively (and my preference) is just say:
if (anchors.find(identifier) != anchors.end()) {
  ...